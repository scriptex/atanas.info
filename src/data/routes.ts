export enum Routes {
	HOME = '/',
	ABOUT = '/about',
	TIMELINE = '/timeline',
	SKILLS = '/skills',
	PORTFOLIO = '/portfolio',
	STATS = '/stats',
	SLIDES = '/slides',
	VIDEOS = '/videos',
	ARTICLES = '/articles',
	CERTIFICATES = '/certificates',
	SOCIAL = '/social',
	BLOG = '/blog',
	MUSIC = '/music'
}
