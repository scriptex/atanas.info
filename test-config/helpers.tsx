import { act } from 'react-dom/test-utils';
import type { FC } from 'react';
import { render, RenderResult, waitFor } from '@testing-library/react';

import * as shared from '@scripts/shared';

type ComponentType = FC<Readonly<React.ComponentProps<any>>>;

export const test = async (Component: ComponentType): Promise<RenderResult> => {
	let result!: RenderResult;

	await act(async () => {
		result = await waitFor(() => render(<Component />));
	});

	return result;
};

export const snapshotTest = (Component: ComponentType, buttonSelector?: string): void => {
	const { name } = Component;

	jest.spyOn(shared, 'formatDate');

	(shared.formatDate as jest.Mock).mockImplementation(() => 'Mock date');

	afterAll(() => {
		jest.clearAllMocks();
	});

	describe(`${name} component`, () => {
		it(`Should render the ${name} component`, async () => {
			const { asFragment, container } = await test(Component);

			expect(asFragment()).toMatchSnapshot();

			if (buttonSelector) {
				await act(async () => {
					await container.querySelectorAll<HTMLElement>(buttonSelector).forEach(button => {
						if (!button.hasAttribute('download')) {
							button.click();

							expect(asFragment()).toMatchSnapshot();
						}
					});
				});
			}
		});

		it(`Should unmount the ${name} component`, async () => {
			const { unmount, asFragment } = await test(Component);

			unmount();

			expect(asFragment()).toMatchSnapshot();
		});
	});
};

export const mockAudioContext = (): jest.Mock => {
	const mockConnect = jest.fn();

	const mockgetByteFrequencyData = jest.fn();

	const mockcreateMediaElementSource = jest.fn(() => ({
		connect: mockConnect
	}));

	const mockcreateAnalyser = jest.fn(() => ({
		connect: mockConnect,
		frequencyBinCount: [0, 1, 2],
		getByteFrequencyData: mockgetByteFrequencyData
	}));

	const mockcreateOscillator = jest.fn(() => ({
		channelCount: 2
	}));

	const mockChannelSplitterConnect = jest.fn(n => n);

	const mockcreateChannelSplitter = jest.fn(() => ({
		connect: mockChannelSplitterConnect
	}));

	const mockedAudioContext = jest.fn(() => ({
		createAnalyser: mockcreateAnalyser,
		createOscillator: mockcreateOscillator,
		createChannelSplitter: mockcreateChannelSplitter,
		createMediaElementSource: mockcreateMediaElementSource
	}));

	return mockedAudioContext;
};

export const mockFetch = (value: Record<string, string | boolean>): void => {
	global.fetch = jest.fn().mockImplementation(() =>
		Promise.resolve({
			json: () => Promise.resolve(value)
		})
	);
};
