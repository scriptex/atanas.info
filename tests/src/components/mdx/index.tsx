import { Routes } from '@data/routes';
import { MDX as Mdx } from '@components';
import { snapshotTest } from '@test-config/helpers';

snapshotTest(
	() => <Mdx id="test" back={Routes.BLOG} title="Test" image="/images/test.jpg" content="Test content" />,
	undefined,
	'MDX'
);
