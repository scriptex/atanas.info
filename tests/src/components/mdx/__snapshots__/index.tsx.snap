// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unknown component Should render the Unknown component 1`] = `
<DocumentFragment>
  <section
    class="c-section c-section--test"
    id="test"
  >
    <div
      class="o-shell"
    >
      <header
        class="c-section__header"
      >
        <h2>
          Test
        </h2>
      </header>
      <div
        class="lines"
      >
        <div
          class="line line--vertical"
        />
        <div
          class="line line--vertical"
        />
        <div
          class="line line--vertical"
        />
      </div>
      <div
        class="lines"
      >
        <div
          class="line line--horizontal"
        />
        <div
          class="line line--horizontal"
        />
        <div
          class="line line--horizontal"
        />
      </div>
      <img
        alt=""
        loading="lazy"
        src="/images/test.jpg"
      />
      <div
        class="c-blog-post"
      >
        <div
          class="c-blog-post__content"
        >
          Test content
        </div>
        <div
          id="disqus_thread"
        />
      </div>
      <div
        class="c-section__actions"
      >
        <button
          class="c-btn"
          type="button"
        >
          Get in touch
        </button>
        <ul
          class="c-section__actions-list"
        >
          <li>
            <button
              class="c-btn c-section__actions-close"
              type="button"
            >
              Close
            </button>
          </li>
          <li>
            <button
              class="c-btn"
              type="button"
            >
              <svg
                class="c-section__actions-icon"
              >
                <use
                  xlink:href="#svg-email"
                />
              </svg>
              <span>
                Contact me
              </span>
              <strong>
                directly
              </strong>
            </button>
          </li>
          <li>
            <a
              class="c-btn"
              href="https://app.usebraintrust.com/talent/782/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="c-section__actions-icon"
              >
                <use
                  xlink:href="#svg-braintrust"
                />
              </svg>
              <span>
                Hire me on
              </span>
              <strong>
                Braintrust
              </strong>
              <svg
                class="c-svg-external-link"
              >
                <use
                  xlink:href="#svg-external-link"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              class="c-btn"
              href="https://www.toptal.com/resume/atanas-atanasov"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="c-section__actions-icon"
              >
                <use
                  xlink:href="#svg-toptal"
                />
              </svg>
              <span>
                Hire me on
              </span>
              <strong>
                Toptal
              </strong>
              <svg
                class="c-svg-external-link"
              >
                <use
                  xlink:href="#svg-external-link"
                />
              </svg>
            </a>
          </li>
        </ul>
        <br
          class="visible-xs-block"
        />
        <a
          class="c-btn"
          href="/blog"
        >
          Go back
        </a>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`Unknown component Should unmount the Unknown component 1`] = `<DocumentFragment />`;
